File: Array/Java/217_Contains_Duplicate.java
//Hash Table(Time:O(n) Space:O(n))
class Solution {
	public boolean containsDuplicate(int[] nums) {
		set<Intager> set = new HashSet<>(nums.length);
		for(int num: nums) {
			if(set.contains(num))
				return true;
			set.add(num);
		}
		return false;
	}
}


//Sorting(Time:O(nlogn) Space:O(1))
class Solution {
	public boolean containsDuplicate(int[] nums) {
		Arrays.sort(nums);
		for(int i = 0; i < nums.length - 1; i++) {
			if(nums[i] == nums[i + 1])
				return true;
		}
		return false;
	}
}

File: Array/C++/217_Contains_Duplicate.cpp
//Hash Table(unordered_set) O(N)
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> set;
        
        for(int i = 0; i < nums.size(); i++) {
            if(set.find(nums[i]) != set.end()) {
                return true;
            }
            set.insert(nums[i]);
        }
        return false;
    }
};

//Sorting O(NlogN)
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        sort(nums.begin(), nums.begin() + nums.size());
        for(int i = 1; i < nums.size(); i++) {
            if(nums[i - 1] == nums[i]) {
                return true;
            }
        }
        return false;
    }    
};

//Brute Force O(N^2)
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        for(int i = 0; i < nums.size(); i++) {
            for(int j = i + 1; j < nums.size(); j++) {
                if(nums[i] == nums[j]) {
                    return true;
                }
            }
        }
        return false;
    }
};